File: cards_backend/src/main/java/de/fantjastisch/cards_backend/card/repository/Card.java
package de.fantjastisch.cards_backend.card.repository;

import de.fantjastisch.cards_backend.card.Link;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import java.util.Set;
import java.util.UUID;

/**
 * Diese Klasse stellt das Modell einer Karteikarte dar, zur internen Kommunikation mit der Datenbank.
 *
 * @author Tamari Bayer, Freja Sender
 */
@Builder
@Data // getter, setter, toString
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Card {
    UUID id;
    String question;
    String answer;
    String tag;
    // use Set to avoid duplicates
    Set<UUID> categories;
    // use Set to avoid duplicates
    private Set<Link> links;


}

File: cards_android_app/app/src/main/java/de/fantjastisch/cards/card/Card.kt
package de.fantjastisch.cards.card

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "cards")
data class Card(
    @ColumnInfo(name = "question") val question: String,
    @ColumnInfo(name = "answer") val answer: String,

    @PrimaryKey val id: UUID = generateCardId()
)

fun generateCardId() = UUID.randomUUID()